pkg/api/validation/validation.go

```diff
@@ -183,6 +183,13 @@ func ValidateEndpointsName(name string, prefix bool) (bool, string) {
 	return NameIsDNSSubdomain(name, prefix)
 }
 
+// ValidateEventName can be used to check whether the given event name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidateEventName(name string, prefix bool) (bool, string) {
+	return NameIsDNSSubdomain(name, prefix)
```

Comment:
Any 3rd party event sender is potentially broken.  We need to have a validation, but we should announce this as backward incompatible change.